src/main/kotlin/com/sourcegraph/cody/sidebar/WebThemeController.kt

```diff
@@ -0,0 +1,42 @@
+package com.sourcegraph.cody.sidebar
+
+import com.intellij.openapi.application.invokeLater
+import com.sourcegraph.config.ThemeUtil
+import java.awt.Color
+import javax.swing.UIManager
+
+class WebThemeController {
+  private var themeChangeListener: ((WebTheme) -> Unit)? = null
+
+  init {
+    UIManager.addPropertyChangeListener { event ->
+      if (event.propertyName == "lookAndFeel") {
+        invokeLater { themeChangeListener?.invoke(getTheme()) }
+      }
+    }
+  }
+
+  fun setThemeChangeListener(listener: (WebTheme) -> Unit) {
+    themeChangeListener = listener
+  }
+
+  fun getTheme(): WebTheme {
+    val themeVariables = mutableMapOf<String, String>()
```

Comment:
Could be also a bit more concise in a functional style:

```
  val themeVariables = UIManager.getDefaults()
      .filterValues { it is Color }
      .mapKeys { toCSSVariableName(it.key.toString()) }
      .mapValues { toCSSColor(it.value as Color) }
```

I asked Cody to rewrite it and it get it perfectly on the first tyr!

Reply:
fixed